metaflow/plugins/aip/aip_metaflow_step.py

```diff
@@ -35,23 +36,49 @@ def _write_card_artifacts(
     passed_in_split_indexes: str,
     run_id: str,
 ):
+    """
+    Pull card artifacts from datastore and add them to the Argo artifact output path.
+    Cards should already be uploaded to the datastore as part of the @card decorator.
+    No exception is thrown: cards are available to view through Metaflow UI or CLI even if this function fails.
+    """
+
     task_id_template: str = f"{task_id}.{passed_in_split_indexes}".strip(".")
     pathspec = f"{flow_name}/{run_id}/{step_name}/{task_id_template}"
 
-    cards: List[Card] = list(get_cards(pathspec))
+    retry_limit = 3
+    cards: List[Card] = []
+    for attempt in range(retry_limit):
+        try:
+            cards: List[Card] = list(get_cards(pathspec))
+        except:
+            if attempt == retry_limit - 1:  # Last attempt failed
+                logging.exception(
+                    f"Failed to get cards from Metaflow backend for pathspec {pathspec}"
+                    "Please view cards through Metaflow UI, or refer to https://docs.metaflow.org/metaflow/visualizing-results/effortless-task-inspection-with-default-cards#accessing-cards-via-an-api"
+                )
+            continue
```

Comment:
if we retry calling the metaflow metadata store, we should put a sleep in between retries, even a of 1s or 0.5 would make a difference.